site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,264 @@
++++
+title = "Tenant API"
+weight = 410
++++
+
+The *Tenant API* is used to make Hono aware of the in the system available tenants. A tenant is a logical entity, which groups together a set of devices and consists of meta information which should be used by other Hono components. The meta information is used for example to determine if a given tenant is allowed to use a certain protocol adapter or if devices are required to use authentication.
+<!--more-->
+
+Note, however, that in real world applications the tenant information will probably be kept and managed by an existing *system of record*, using e.g. a database for persisting the data. The Tenant API accounts for this fact by means of defining only the [Get Tenant Information]({{< relref "#get-tenant-information" >}}) operation as *mandatory*, i.e. this operation is strictly required by a Hono instance for it to work properly, whereas the remaining operations are defined as *optional* from a Hono perspective.
+
+The Tenant API is defined by means of AMQP 1.0 message exchanges, i.e. a client needs to connect to Hono using an AMQP 1.0 client in order to invoke operations of the API as described in the following sections.
+
+# Preconditions
+
+The preconditions for performing any of the operations are as follows:
+
+1. Client has established an AMQP connection with Hono.
+2. Client has established an AMQP link in role *sender* with Hono using target address `tenant`. This link is used by the client to send tenant commands to Hono.
+3. Client has established an AMQP link in role *receiver* with Hono using source address `tenant/${reply-to}` where *reply-to* may be any arbitrary string chosen by the client. This link is used by the client to receive responses to the tenant api requests it has sent to Hono. This link's source address is also referred to as the *reply-to* address for the request messages.
+
+This flow of messages is illustrated by the following sequence diagram (showing the AMQP performatives):
+
+![Tenant message flow preconditions](../tenant_ConnectToTenant.png)
+
+# Operations
+
+The operations described in the following sections can be used by clients to manage tenant information. In real world scenarios the provisioning of tenants will most likely be an orchestrated process spanning multiple components of which Hono will only be one.
+
+Conducting and orchestrating the overall provisioning process is not in scope of Hono. However, Hono's Tenant API can easily be used as part of such an overall provisioning process.
+
+
+## Add Tenant
+
+Clients use this command to initially *add* information about a new tenant to Hono.
+
+This operation is *optional*, implementors of this API may provide other means for adding tenant information, e.g. a RESTful API or a configuration file.
+
```

Comment:
...for adding tenant, e.g. (skip information)

Reply:
adapted